Sources/Afluent/SequenceOperators/ThrottleSequence.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  ThrottleSequence.swift
+//
+//
+//  Created by Trip Phillips on 2/12/24.
+//
+
+import Foundation
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension AsyncSequences {
+    public struct Throttle<Upstream: AsyncSequence, C: Clock>: AsyncSequence {
+        public typealias Element = Upstream.Element
+        let upstream: Upstream
+        let interval: C.Duration
+        let clock: C
+        let latest: Bool
+        
+        class IntervalEvents {
+            var hasSeenFirstElement: Bool
+            var hasStartedInterval: Bool
+            var firstElement: Element?
+            var latestElement: Element?
+            var startInstant: C.Instant?
+            
+            private let lock = NSRecursiveLock()
+            
+            init(hasSeenFirstElement: Bool = false,
+                 hasStartedInterval: Bool = false,
+                 firstElement: Element? = nil,
+                 latestElement: Element? = nil,
+                 startInstant: C.Instant? = nil) {
+                self.hasSeenFirstElement = hasSeenFirstElement
+                self.hasStartedInterval = hasStartedInterval
+                self.firstElement = firstElement
+                self.latestElement = latestElement
+                self.startInstant = startInstant
+            }
+            
+            func updateHasSeenFirstElement() {
+                lock.protect {
```

Comment:
So what I don't like here is your actual getter/setter isn't protected, only your update methods are. I think you should change this whole class so that the getter/setter is lock protected with `_` private properties that do the storage. There's a lot of room here for maintainers to make a mistake and I think it's just because this got refactored from an actor.

Reply:
Sorry if this conversation already happened, but is there a reason why this can't be an actor? I don't see it used anywhere that's not already async

Reply:
Answered offline: actors are inefficient for high-performance contexts